### PR TITLE
Respect skipFormatDates when sent as string from JS

### LINKED ISSUE #14016: Processors do not respect type of the parameter that was passed from javascript code
### Summary
Passed boolean parameters do not properly process in processors and getProperty returns strings instead of the proper parameter type. It causes various issues in the manager when condition should work but it doesn't.

### Step to reproduce
Create any combo-box or other extjs component or use existing and pass any boolean parameter into the processor. Like this:
```
baseParams: {
    action: 'source/getList',
    showNone: false,
    streamsOnly: true
}
```

### Observed behavior
In the processor `$this->getProperty('showNone')` returns string instead of boolean variable. If to try to cast the type to boolean it will be always true because the string is not empty (`(bool)'false'` and `(bool)'true'` will be always true for both).

### Expected behavior
`$this->getProperty('myvar')` should return proper boolean value as it was passed into the processor.

### Environment
2.7.0-dev. 3.x also, and all previous too.

### PR DESCRIPTION
**Pull request by Ibochkarev**
### What does it do?

Fixes `Resource/Get` so the `skipFormatDates` request parameter is evaluated with the existing `modX::paramValueIsTrue()` helper instead of raw `getProperty()` checks.

When the manager sends `skipFormatDates=false` as a string from JavaScript, the previous condition still treated it as truthy and skipped date formatting incorrectly. The processor now only skips formatting when the parameter is explicitly true (`true`, `'true'`, `1`, `'1'`).

Also fixes `formatDates()` to check `publishedon` directly instead of `!empty($resourceArray)`.

No changes to `Processor::getProperty()` or new processor helpers.

### Why is it needed?

JavaScript/ExtJS sends boolean values as strings (`"true"` / `"false"`). In `Resource/Get`, `getProperty('skipFormatDates')` made `"false"` truthy, so formatted dates were skipped when they should not have been.

Using the existing `paramValueIsTrue()` method keeps behavior consistent with the rest of MODX and avoids broader processor API changes.

### How to test

1. Open a resource in the manager that triggers `Resource/Get` with `skipFormatDates=false` (string).
2. Confirm `pub_date`, `unpub_date`, and `publishedon` are still formatted for display.
3. Call the processor with `skipFormatDates=true` and confirm date fields are returned unformatted.
4. Verify resources with empty `publishedon` still return an empty string after formatting.

### Related issue(s)/PR(s)

Fixes #14016

### CODE REVIEW

**Comment by Mark-H** [comment]
I'm mildly worried this may have unintentional far-reaching consequences in processors that have been doing its own normalisation based on strings that is now going to break.

**Comment by Mark-H** [comment]
> Introduce a new method like getBooleanProperty() that performs normalization

I think this suggestion makes *a lot* of sense given the potential impact this may have on core and extras. Copilot already highlighted a few cases in the codebase that break.

**Comment by Ibochkarev** [comment]
@Mark-H please re-run review copilot. Thanks

**Comment by biz87** [comment]
## Code Review

**Summary**

Adds `Processor::getBooleanProperty($k, $default)` to normalize string boolean values (`"true"`, `"false"`, `"on"`, `"off"`, `"1"`, `"0"`, `"yes"`, `"no"`) to proper PHP booleans. Existing `getProperty()` is unchanged. Includes comprehensive PHPUnit tests covering all edge cases (case insensitivity, trimming, native booleans, missing keys, unknown strings).

**Assessment**

Clean, non-breaking addition. The `match` expression (PHP 8.0+) is appropriate for the project's PHP 8.1+ requirement. Test coverage is thorough — 13 test methods covering all branches. The method properly falls through to `$default` for unrecognized strings and empty values, which is the correct behavior. No existing code is modified beyond adding the new method and test file.

**Verdict**: Approve

File: core/src/Revolution/Processors/Resource/Get.php [hunk 1]
```diff
@@ -30,8 +31,7 @@ public function process()
     {
         $resourceArray = $this->object->toArray();
         $resourceArray['canpublish'] = $this->modx->hasPermission('publish_document');
-        if (!$this->getProperty('skipFormatDates') ||
-            ($this->getProperty('skipFormatDates') && $this->getProperty('skipFormatDates') == 'false')) {
+        if (!$this->getBooleanProperty('skipFormatDates', false)) {
```
Comment (smg6511) [inline core/src/Revolution/Processors/Resource/Get.php]: There's an already existing method that can be used, so no need for a new one:
```suggestion
        $skipFormatDates = $this->modx->paramValueIsTrue($this->getProperties(), 'skipFormatDates');
        if (!$skipFormatDates) {
```

**Review by smg6511 — CHANGES_REQUESTED** [review]
I'm of the mind that we don't need to be covering the boolean-ish values of "yes" and "on," particularly in the context of the manager backend where we have control over how boolean values are passed. With this in mind, we already have a method for dealing with typical string booleans.

**Comment by smg6511** [comment]
@Ibochkarev - Just a heads up that since `getBooleanProperty` isn't needed you can drop the method and its associated unit tests.

**Comment by Ibochkarev** [comment]
> @Ibochkarev - Just a heads up that since `getBooleanProperty` isn't needed you can drop the method and its associated unit tests.

Ok

**Comment by smg6511** [comment]
@Ibochkarev - This one will be ready once you remove the three unneeded files, leaving just the changed `Get.php`.

**Comment by Ibochkarev** [comment]
> @Ibochkarev - This one will be ready once you remove the three unneeded files, leaving just the changed `Get.php`.

Done

**Review by smg6511 — APPROVED** [review]
Good, safe change after simplification and use of pre-existing method.

File: core/src/Revolution/Processors/Processor.php [hunk 2]
```diff
@@ -214,13 +214,24 @@ public function run()
...
+        $value = array_key_exists($k, $this->properties) ? $this->properties[$k] : $default;
+        if (is_string($value)) {
+            $normalized = strtolower(trim($value));
+            return match ($normalized) {
+                'true' => true,
+                'false' => false,
+                default => $value,
+            };
+        }
+        return $value;
```
[the remark this answers is not in the document]
Comment (smg6511) [reply core/src/Revolution/Processors/Processor.php]: File and method removed, no longer applicable.

**Comment by opengeek** [comment]
I think the PR title/description for this is outdated. Can someone review and update this to accurately reflect what this PR actually does?

**Comment by Ibochkarev** [comment]
@opengeek Updated the PR title and description to match the final scope: a targeted fix in `Resource/Get` using `modX::paramValueIsTrue()` for `skipFormatDates`, with no changes to `Processor::getProperty()` or new processor helpers.
